docs/kilt-baseline-replication.md

```diff
@@ -0,0 +1,136 @@
+# Replication guide KILT
+
+## Setup env
+
+```bash
+# Create a virtual env
+conda create -n kilt37 -y python=3.7 && conda activate kilt37
+
+# Get pyserini scripts and install package
+git clone https://github.com/castorini/pyserini.git
+pip install pyserini
+
+# Get KILT scripts and install package
+git clone https://github.com/facebookresearch/KILT.git
+cd KILT
+pip install -r requirements.txt
+pip install .
+
+# Get KILT queries and gold data
+mkdir data
+python scripts/donwload_all_kilt_data.py
+python scripts/get_triviaqa_input.py
+cd ..
+
+# Get the KILT knowledge source / wikipedia dump (34.76GiB)
+wget http://dl.fbaipublicfiles.com/KILT/kilt_knowledgesource.json
+
+# Get NLTK dependencies
+python -m nltk.downloader punkt
+python -m nltk.downloader stopwords
+```
+
+## Index the knowledge source
+
+Convert to passage or document level JSONL format indexable by Pyserini (takes 1-2hr if bigrams is enabled):
+
+```bash
+# We'll split it in multiple files to make this faster
+mkdir kilt_knowledge_split
+cd kilt_knowledge_split
+split -l500000 ../kilt_knowledgesource.json kilt_ks.
+cd ..
+
+# Run conversion script for each file
+mkdir dump
+```
+
+For document level:
+```bash
+mkdir dump/doc_dump
+for filename in kilt_knowledge_split/kilt_ks.??; do
+    [ -e "$filename" ] || continue
+    nohup python pyserini/scripts/kilt/convert_kilt_to_document_jsonl.py \
+        --input "$filename" \
+        --output dump/doc_dump/$(basename "$filename") \
+        --bigrams --stem \
+        --flen 500000 \
+        > nohup_$(basename "$filename").out &
+done
+
+# Once it's done, convert back into 1 file:
+cat dump/doc_dump/kilt_ks.?? > dump/doc_dump/dump.jsonl
+rm dump/doc_dump/kilt_ks.??
+# Sanity check (should give the same # of lines):
+wc -l kilt_knowledgesource.json
+wc -l dump/doc_dump/dump.jsonl
+```
+
+For passage level:
+```bash
+mkdir dump/passage_dump
+for filename in kilt_knowledge_split/kilt_ks.??; do
+    [ -e "$filename" ] || continue
+    nohup python pyserini/scripts/kilt/convert_kilt_to_passage_jsonl.py \
+        --input "$filename" \
+        --output dump/passage_dump/$(basename "$filename") \
+        --sections --bigrams --stem \
+        --flen 500000 \
+        > nohup_$(basename "$filename").out &
+done
+
+# Once it's done, convert back into 1 file:
+cat dump/passage_dump/kilt_ks.?? > dump/passage_dump/dump.jsonl
+rm dump/passage_dump/kilt_ks.??
+``` 
+
+Index into Anserini (about 1hr). These instructions are the same whether you are using document or passage level index (just change the path accordingly).:
+
+```bash
+nohup python -m pyserini.index -collection JsonCollection -generator DefaultLuceneDocumentGenerator \
+ -threads 40 -input <path to dump.jsonl e.g. dump/doc_dump/> \
```

Comment:
put in the actual file name here...

Reply:
I was planning on updating these instructions with specific paths once we had a baseline that beats DrQA, but I was quite surprised to see that bigrams actually reduced the precision for the passage level index.